keps/sig-api-machinery/20190226-network-proxy.md

```diff
@@ -0,0 +1,577 @@
+We will add configuration as to which proxy an etcd client should use. 
+(https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go) 
+This will add an extra process hop to our main scaling axis. 
+We will scale test the impact and publish the results. As a precaution
+we will add an extra network configuration 'etcd' separate from ‘master’.
+Etcd requests can be configured separately from the rest of 'master'.  
+- **Pods/Exec**, **Pods/Proxy**, **Pods/Portforward**, **Pods/Attach**, **Pods/Log**
+Pod requests (and pod sub-resource requests) are meant for the cluster 
+and will be routed based on the ‘cluster’ NetworkContext.
+- **Nodes/Proxy**
+Node requests (and node sub-resource requests) are meant for the cluster 
+and will be routed based on the ‘cluster’ NetworkContext.
+- **Services/Proxy**
+Service requests (and service sub-resource requests) are meant for the cluster 
+and will be routed based on the ‘cluster’ NetworkContext.
+- **Admission Webhooks**
+Admission webhooks can be destined for a service or a URL. 
+If destined for a service then the service rules apply (send to 'cluster'). 
+If destined for a URL then we will use the ‘master’ NetworkContext. 
+- **Aggregated API Server (and OpenAPI requests for aggregated resources)**
+Aggregated API Servers can be destined for a service or a URL. 
+If destined for a service then the service rules apply. 
+If destined for a URL then we will use the ‘master’ NetworkContext.
+- **Authentication, Authorization and Audit Webhooks**
+These Webhooks use a kube config file to determine destination. 
+Given that we use a ‘master’ NetworkContext.
+- **ImagePolicyWebhook**
+The image policy webhook uses a kube config file to determine destination. 
+Given that we use a ‘master’ NetworkContext.
+- **KMS GRPC Service**
+KMS connects with an ‘endpoint’ (not the resource) via gRPC. 
+The service at the endpoint provides the secret information for use in encryption. 
+This is not a user space configurable system. 
+Given that we use a ‘master’ NetworkContext.
+
+### Testing the Solution
+
+We will test using a network namespace to partition the KAS from the test nodes. 
+It is then impossible to connect directly from the KAS to the test nodes. 
+This ensures that the proxy must be used for logs, exec, port forward, aggregation and webhooks. 
+We run with this configuration and a direct configuration for these specific features. 
+This ensures that the solution works and will continue to work.
+
+### Security
+
+One motivation for network proxy is providing a mechanism to secure 
+https://groups.google.com/d/msg/kubernetes-security-announce/tyd-MVR-tY4/tyREP9-qAwAJ. 
+This, in conjunction with a firewall or other network isolation, fixes the security concern. 
+
+### Implementation Details/Notes/Constraints
+
+You may want to check the original design doc for alternatives and futures considered. https://goo.gl/qiARUK.
```

Comment:
If we're linking out to another doc here we should mention what mailing list folks have to join to get read access.

Reply:
kubernetes-dev@googlegroups.com should now have comment access. Will add that to the KEP.